docs/manual/src/Upgrading.md

```diff
@@ -0,0 +1,35 @@
+# v0.28.x -> v0.29.x
+
+## Custom types
+
+Custom types are now implemented using a macro, rather than implementing the
+`UniffiCustomTypeConverter` trait.
+
+Before:
+
+```rust
+impl UniffiCustomTypeConverter for MyType {
+    type Builtin = UniFFIBuiltinType;
+
+    fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
+        ...
+    }
+
+    fn from_custom(obj: Self) -> Self::Builtin {
+        ...
+    }
+}
+```
+
+After:
+
+```
+uniffi::custom_type!(MyType, UniFFIBuiltinType, {
+    try_from_existing: |val| { ... },
+    into_existing: |obj| { ... },
+})
+```
+
+The custom_type macro is more flexible than the old system.  For example, the `try_from_existing` and
+`into_existing` can be omitted in many cases, and will use the `TryInto` and `From` traits by default.
```

Comment:
Reading the docs below, it looks like we might also use `Into` and `TryFrom` depending on the direction/context. If so, could you please tweak this? In particular, this sentence almost seems like the reverse of what's sensible (ie, that we have a `try_from` method, which uses `From` if omitted, and we have an `into` method which, if omitted, uses `TryInto` appears wrong and inconsistent).

(If I'm not wrong about that, the changes make here could make the text quite vague, which I think is OK if the alternative makes things appear more restrictive than it is)

Reply:
Thinking about this some more, I think we should spell out the exact trait and always use the into version.  So this should say "For example, the `try_from_existing` and `into_existing` can be omitted in many cases, and will use the `TryInto<Custom>` and `Into<Existing>` traits by default."  The reason for this is that Into is auto-derived from From, but not the other way around.

Maybe we should rename `try_from_existing` to `try_into_custom` to match this?